src/main/java/com/example/finalproject/service/RecipeService.java

```diff
@@ -0,0 +1,74 @@
+package com.example.finalproject.service;
+
+import com.example.finalproject.model.Category;
+import com.example.finalproject.model.CategoryName;
+import com.example.finalproject.model.Ingredient;
+import com.example.finalproject.model.Recipe;
+import com.example.finalproject.repository.RecipeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class RecipeService {
+
+    private RecipeRepository recipeRepository;
+
+    @Autowired
+    public RecipeService(RecipeRepository recipeRepository) {
+        this.recipeRepository = recipeRepository;
+    }
+
+    public void persistRecipe(Recipe recipeForm, Ingredient ingredientForm, Category category) {
+        Category categoryFromDb = findCategoryByName(category.getCategoryName()).get();
+
+        Recipe recipe = new Recipe(recipeForm.getTitle(), recipeForm.getDescription(), recipeForm.getTime(), recipeForm.getDifficulty(), recipeForm.getPreparation());
+        Ingredient ingredient = new Ingredient(ingredientForm.getName(), ingredientForm.getQuantity(), ingredientForm.getUnit(), ingredientForm.getAdditionalInfo());
+        recipe.addIngredient(ingredient);
+        recipe.setCategory(categoryFromDb);
+        recipeRepository.save(recipe);
+    }
+
+    public void updateRecipe(Recipe recipeForm) {
+        recipeRepository.updateAllFields(recipeForm.getId(),recipeForm.getTitle(), recipeForm.getDescription(), recipeForm.getTime(),
```

Comment:
Tak jak napisałem wyżej - jeżeli Recipe ma w polu ID odpowiednie id to wystarczy zrobić recipeRepository.save()